source/steamshim/src/child/child.cpp

```diff
@@ -122,15 +100,55 @@ static STEAMSHIM_Event* ProcessEvent(){
     return &event;
 }
 
+static bool setEnvironmentVars(PipeType pipeChildRead, PipeType pipeChildWrite)
+{
+    char buf[64];
+    snprintf(buf, sizeof (buf), "%llu", (unsigned long long) pipeChildRead);
+    if (!setEnvVar("STEAMSHIM_READHANDLE", buf))
+        return false;
+
+    snprintf(buf, sizeof (buf), "%llu", (unsigned long long) pipeChildWrite);
+    if (!setEnvVar("STEAMSHIM_WRITEHANDLE", buf))
+        return false;
+
+
+
+    return true;
+} 
+
 extern "C" {
-  int STEAMSHIM_init(void)
+  int STEAMSHIM_init(bool isclient)
   {
+
+
+    PipeType pipeParentRead = NULLPIPE;
+    PipeType pipeParentWrite = NULLPIPE;
+    PipeType pipeChildRead = NULLPIPE;
+    PipeType pipeChildWrite = NULLPIPE;
+    ProcessType childPid;
+
+    if (isclient)
+        setEnvVar("STEAMSHIM_ISCLIENT", "1");
+
+
+    if (!createPipes(&pipeParentRead, &pipeParentWrite, &pipeChildRead, &pipeChildWrite))
+        fail("Failed to create application pipes");
+    else if (!setEnvironmentVars(pipeChildRead, pipeChildWrite))
+        fail("Failed to set environment variables");
+    else if (!launchChild(&childPid, STEAM_BLOB_LAUNCH_NAME))
+        fail("Failed to launch application");
```

Comment:
Not sure if bringing down the game in any of these instances is a good idea.
Better to have a game without functioning steam than no game, right?